Source/Dafny/AST/DafnyAst.cs

```diff
@@ -13,14 +13,60 @@
 using System.Diagnostics.Contracts;
 using System.Numerics;
 using System.Linq;
-using Microsoft.Boogie;
 using System.Diagnostics;
 using System.Threading;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
+  public interface IToken : Microsoft.Boogie.IToken {
+    // Inherited for now.
+    /*
+    int kind { get; set; }
+    string filename { get; set; }
+    int pos { get; set; }
+    int col { get; set; }
+    int line { get; set; }
+    string val { get; set; }
+    bool IsValid { get; }*/
+    string leadingTrivia { get; set; }
```

Comment:
These should start with a capital right?

Reply:
Yes they should like in most interfaces, if we wrote this interface from scratch.
However, since I'm inheriting from Boogie tokens, I keep their style.

Reply:
I think you can make all of them capital case by using an explicit interface implementation for the Boogie members, like this:

```c#
int Boogie.IToken.pos {
  get => Pos;
  set => Pos = value;
}

public abstract int Pos { get; set; }
```

Reply:
That can be done in another PR, right?

Reply:
In another PR, we could do this change AND also make sure we don't use the Boogie's lowercase interface, but only Dafny's one.

Reply:
I don't like approving new lines of code that don't follow our coding standards. 

How about you introduce the properties like this:

```c#
interface IToken : Boogie.IToken {
  int Boogie.IToken.pos {
    get => pos;
    set => pos = value;
  }

  int pos { get; set; }
}
```

and then rename `pos` to `Pos` using the IDE, so it's almost no work on the editing side?

Removing the inheritance on `Boogie.IToken` doesn't seem so valuable right now. I think we can shelf that refactoring for the time being.

Reply:
Ok. I did try the refactoring. + manually on the frames files (Scanner/Parser). But it's just impossible: Coco/R uses the given class Token and believes that the fields should be lowercase (and in the generated code they are treated as such). There is no way I can modify the tool to change that.
https://ssw.jku.at/Research/Projects/Coco/

Again, the code we are merging was already part of a dependency, so not doing this refactoring is the statu quo, and, based on the experience above, makes more sense as of today.

Reply:
Okay, let's leave it like this then. Do you know which fields Coco/R is targeting? Could you put comments on them to explain why they're lowercase?